pkg/storage/fs/ocis/path.go

```diff
@@ -132,30 +145,34 @@ func (pw *Path) readRootLink(root string) (node *Node, err error) {
 	return
 }
 
-// RootNode returns the root node of a tree,
-// taking into account the user layout if EnableHome is true
+// RootNode returns the root node of a tree
 func (pw *Path) RootNode(ctx context.Context) (node *Node, err error) {
-	var root string
-	if pw.EnableHome && pw.UserLayout != "" {
-		// start at the users root node
-		u := user.ContextMustGetUser(ctx)
-		layout := templates.WithUser(u, pw.UserLayout)
-		root = filepath.Join(pw.Root(), "users", layout)
+	return &Node{
+		pw:       pw,
+		ID:       "root",
+		Name:     "",
+		ParentID: "",
+		Exists:   true,
+	}, nil
+}
 
-	} else {
-		// start at the storage root node
-		root = filepath.Join(pw.Root(), "nodes/root")
+// HomeNode returns the home node of a user
+func (pw *Path) HomeNode(ctx context.Context) (node *Node, err error) {
+	if !pw.EnableHome {
+		return nil, errtypes.NotSupported("ocisfs: home supported disabled")
 	}
 
-	// The symlink contains the nodeID
-	node, err = pw.readRootLink(root)
-	if err != nil {
+	if node, err = pw.RootNode(ctx); err != nil {
 		return
 	}
-	return
-}
+	u := user.ContextMustGetUser(ctx)
+	layout := templates.WithUser(u, pw.UserLayout)
 
-// Root returns the root of the storags
-func (pw *Path) Root() string {
-	return pw.root
+	segments := strings.Split(layout, "/")
+	for i := range segments {
```

Comment:
didn't we have a similar loop/logic in NodeInfo that we can reuse ?

Reply:
introduced `func (pw *Path) WalkPath(ctx context.Context, r *Node, p string, f func(ctx context.Context, n *Node) error) (*Node, error)` 
using it in three other places